src/sagemaker/model_card/model_card.py

```diff
@@ -898,8 +1152,13 @@ def __init__(
         self.training_details = training_details
         self.evaluation_details = evaluation_details
         self.additional_information = additional_information
+        self.model_package_details = self._from_model_package(model_package)
```

Comment:
model_package should be assigned directly i.e. 
```
self.model_package_details = model_package
```
but we will call `self._from_model_package(model_package)` in the beginning of the __init__() to auto-populate the training and evaluation details.

Reply:
we could do `self.model_package_details = self._from_model_package(model_package)` then the `self._from_model_package`  will return the `model_package`

I don't think it is a good practice though because the function name `_from_model_package` means to get something from model_package which should either return nothing or return something derived from model_package data. Good to separate two steps

Reply:
Oh yeah you are right, my bad. `model_package_details` only contains the model package metadata information not the training details or evaluation details.